flight/mcu/lib/teensyShared/rocketFSM.cpp

```diff
@@ -127,179 +134,36 @@ void rocketFSM::tickFSM() {
+                chVTGetSystemTime() - pointer_struct->rocketTimers.burnout_time;
 
             // If the low acceleration lasts long enough, coast is detected
-            if (TIME_I2MS(rocketTimers.coast_timer) > coast_time_thresh) {
+            if (TIME_I2MS(pointer_struct->rocketTimers.coast_timer) >
+                coast_time_thresh) {
                 pointer_struct->sensorDataPointer->rocketState_data
                     .rocketState = STATE_COAST;
             }
 
             break;
 
         case STATE_COAST:
+            // Serial.println("COAST");
+            pointer_struct->rocketTimers.coast_timer =
+                chVTGetSystemTime() - pointer_struct->rocketTimers.burnout_time;
 
-            rocketTimers.coast_timer =
-                chVTGetSystemTime() - rocketTimers.burnout_time;
-
-            if (TIME_I2MS(rocketTimers.coast_timer) >
-                coast_to_apogee_time_threash) {
+            if (TIME_I2MS(pointer_struct->rocketTimers.coast_timer) >
+                coast_to_apogee_time_thresh) {
                 pointer_struct->sensorDataPointer->rocketState_data
                     .rocketState = STATE_APOGEE;
+            } else {
+                // Serial.println("Still in coast");
             }
 
-            // // if velocity <= 0
-            // if (pointer_struct->sensorDataPointer->state_data.state_vz <=
-            //     apogee_thresh) {
-            //     rocketTimers.apogee_time = chVTGetSystemTime();
-            //     pointer_struct->sensorDataPointer->rocketState_data
-            //         .rocketState = STATE_APOGEE_DETECT;
-            // }
-
             break;
-
+        case STATE_APOGEE:
+            // Serial.println("APOGEE");
         default:
             break;
-
-            // we don't have a measure of velocity, so to ensure that active
-            // control works for the whole launch this state will be the final
-            // state
-
-            // case STATE_APOGEE_DETECT:
-            //     // If the negative velocity was too brief, go back to coast
-            //     if (pointer_struct->sensorDataPointer->state_data.state_vz >
-            //         apogee_thresh) {
-            //         pointer_struct->sensorDataPointer->rocketState_data
-            //             .rocketState = STATE_COAST;
-            //         break;
-            //     }
-
-            //     // measure the length of the apogee time
-            //     rocketTimers.apogee_timer =
-            //         chVTGetSystemTime() - rocketTimers.apogee_time;
-
-            //     // If the < 0 velocity lasts long enough, apogee is detected
-            //     if (rocketTimers.coast_timer > apogee_time_thresh) {
-            //         pointer_struct->sensorDataPointer->rocketState_data
-            //             .rocketState = STATE_APOGEE;
-            //     }
-            //     break;
-
-            // case STATE_APOGEE:  //*assumes velocity is <= 0
```

Comment:
We need to get back the more advanced apogee detect, and parachute detects. That's useful. Right now apogee is just a timer but if we get reliable data from state estimation we can use that instead.

Reply:
Planning on making this a task for newcomers (or anyone tbh) next year